README.md

```diff
@@ -150,6 +152,16 @@ const env = cleanEnv(process.env, {
 });
 ```
 
+### TypeScript users
+
+You can use either one of `makeBaseValidator`, `makeExactValidator` and `makeMarkupValidator`
+depending on your use case:
+
+- `makeBaseValidator<BaseT>` when you want the output to be narrowed-down to a subtype of `BaseT` (e.g. `str`).
+- `makeExactValidator<T>` when you want the output to be widened to `T` (e.g. `bool`).
```

Comment:
Could probably use examples of each case (makeValidator vs makeExactValidator) here as I don't think it'll be clear to most readers without diving into the source

Reply:
Let me know if that's enough: 9727612d86e9f4ab003ab51f766e45d98f41d504

Reply:
Couldn't find the comment about formatting again; but fixed it here: ce1007f37601ba51a67fa3718321e404bdc569c6